opendap_protocol/protocol.py

```diff
@@ -53,6 +54,13 @@
 @dataclass
 class Config:
     DASK_ENCODE_CHUNK_SIZE: int = 20e6
+    DASK_CACHE_SIZE: int = 120 * 1024 * 1024  # 120MB
+
+# we load one `DASK_ENCODE_CHUNK_SIZE`-sized block of linearized data
+# in to memory at one go. This may overlap with multiple dask chunks
+# so lets cache those chunks since we might come back to them.
+cache = Cache(Config.DASK_CACHE_SIZE)
```

Comment:
Shouldn't this be configured at the server level? That is what we do

Reply:
Now I think we should apply the cache more locally in that loop in `dods_encode`. We want to cache aggressively when we have multiple batches to stream out for a *single* request from a *single* array.

I think the more global server cache is appropriate for a less aggressive cache across *multiple* requests.

Perhaps we can pair at some point and just iterate through some options with a benchmark problem.

Reply:
Also seems like a good place to stick in a bit of async: compute the next iteration while streaming out the current iteration.